core.py

```diff
@@ -357,18 +358,23 @@ def append_targets(targets, container, prefix, total_size, total_cnt):
                 prefix = project['group'] + '/' + project['name'] + '/' + session.get('label', 'untitled') + '/' + acq.get('label', 'untitled')
                 total_size, file_cnt = append_targets(targets, acq, prefix, total_size, file_cnt)
         log.debug(json.dumps(targets, sort_keys=True, indent=4, separators=(',', ': ')))
-        filename = 'sdm_' + datetime.datetime.utcnow().strftime('%Y%m%d_%H%M%S') + '.zip'
+        filename = 'sdm_' + datetime.datetime.utcnow().strftime('%Y%m%d_%H%M%S') + '.tar'
         ticket = util.download_ticket('batch', targets, filename, total_size)
         self.app.db.downloads.insert(ticket)
         return {'ticket': ticket['_id'], 'file_cnt': file_cnt, 'size': total_size}
 
     def _archivestream(self, ticket):
-        length = None # FIXME compute actual length
-        z = zipstream.ZipFile(allowZip64=True)
         targets = ticket['target']
-        for filepath, acrpath, _ in targets:
-            z.write(filepath, acrpath)
-        return z, length
+        s = cStringIO.StringIO()
+        z = tarfile.open(mode="w|", fileobj=s)
+        for filepath, arcpath, size in targets:
+            z.add(filepath, arcname=arcpath, recursive=False)
+            yield s.getvalue()
+            s.truncate(0)
+            s.seek(0)
+        z.close()
+        yield s.getvalue()
```

Comment:
Of course, that makes sense. After all, tar is meant for tape and you need to get to the end of the tape sector or whatever that's called.